Adds ReplaySubject and Publisher.share(replay:) (#23)
CombineCommunity/CombineExt: README.md

removeAllDuplicates: 4
```

------

> ### ShareReplay

Reviewer: ```suggestion
### share(replay:)
```

> Similar to [`Publisher.share`](https://developer.apple.com/documentation/combine/publisher/3204754-share), `.share(replay:)` can be used to create a publisher instance with reference semantics. But, it also replays value events to further subscribers.

Reviewer: ```suggestion
Similar to [`Publisher.share`](https://developer.apple.com/documentation/combine/publisher/3204754-share), `.share(replay:)` can be used to create a publisher instance with reference semantics which replays a pre-defined amount of value events to further subscribers.
```

guarantees
```

## Subjects

### ReplaySubject

> A Combine analog to Rx’s [`ReplaySubject` type](http://reactivex.io/documentation/subject.html). It’s similar to a [`CurrentValueSubject`](https://developer.apple.com/documentation/combine/currentvaluesubject) in that it buffers elements, but, it takes it a step further in allowing consumers to specific the number of value events to buffer and replay to future subscribers. Also, it will handle forwarding any completion events after the buffer is cleared upon subscription.

Reviewer: ```suggestion
A Combine analog to Rx’s [`ReplaySubject` type](http://reactivex.io/documentation/subject.html). It’s similar to a [`CurrentValueSubject`](https://developer.apple.com/documentation/combine/currentvaluesubject) in that it buffers values, but, it takes it a step further in allowing consumers to specify the number of values to buffer and replay to future subscribers. Also, it will handle forwarding any completion events after the buffer is cleared upon subscription.
```

```swift

> let subject = ReplaySubject<Int, Never>(maxBufferSize: 3)

Reviewer: Any reason for the `max`? I'd just go `bufferSize`

Author: Updating in the next commit. 👌🏽
